cupy/linalg/norms.py

```diff
@@ -70,14 +75,28 @@ def norm(x, ord=None, axis=None, keepdims=False):
             return abs(x).sum(axis=axis, keepdims=keepdims)
         elif ord is None or ord == 2:
             # special case for speedup
-            s = x ** 2
+            if issubclass(x.dtype.type, numpy.complexfloating):
+                s = abs(x)
+                s *= s
+            else:
+                s = x ** 2
             return cupy.sqrt(s.sum(axis=axis, keepdims=keepdims))
         else:
             try:
                 float(ord)
             except TypeError:
                 raise ValueError("Invalid norm order for vectors.")
-            absx = abs(x).astype('d')
+
+            # Mirror Numpy behavior of casting to double for non-complex
+            # dtypes or reduction over all axes
+            reduce_all_axes = (tuple(sorted(axis)) == tuple(range(nd)) and
```

Comment:
How about moving after L94?

Reply:
Sorry, I don't quite understand what you mean. Do you want me to inline the definition in line 95?

Reply:
Sorry about less information.
```python
cast_dtype = 'd'
if issubclass(x.dtype.type, numpy.complexfloating):
    # Mirror Numpy behavior of casting to double for non-complex
    # dtypes or reduction over all axes
    if keepdims or tuple(sorted(axis)) != tuple(range(nd):
       cast_dtype = x.dtype.char.lower()  # 'D'->'d' and 'F'->'f'
```

Reply:
Thanks for the clarification, I implemented the change.